osu.Game/Overlays/NowPlayingOverlay.cs

```diff
@@ -253,25 +253,43 @@ protected override void Update()
 
         private Action pendingBeatmapSwitch;
 
+        private void setMetadata(WorkingBeatmap beatmap)
+        {
+            if (beatmap?.Beatmap == null) //this is not needed if a placeholder exists
+            {
+                title.Text = @"Nothing to play";
+                artist.Text = @"Nothing to play";
+            }
+            else
+            {
+                BeatmapMetadata metadata = beatmap.Metadata;
+                title.Text = new LocalisedString((metadata.TitleUnicode, metadata.Title));
+                artist.Text = new LocalisedString((metadata.ArtistUnicode, metadata.Artist));
+            }
+        }
+
         private void trackChanged(WorkingBeatmap beatmap, TrackChangeDirection direction = TrackChangeDirection.None)
         {
+            if (Alpha == 0)
+            {
+                setMetadata(beatmap);
+                LoadComponentAsync(new Background(beatmap) { Depth = float.MaxValue }, newBackground =>
```

Comment:
not even worth mentioning but this is a memory / performance ordeal. read comments below to know why

Reply:
How would one see this performance drop? Just got done clicking next track 1000x and didn't notice any increase in memory usage.

Reply:
while the now playing overlay is not present, it will still be loading components in the background, but not process the callbacks (they are scheduled to itself).

they will pile up, causing memory leaks (beatmap references, as mentioned in the comment below) and also add the overhead of doing the `LoadComponentAsync` of the background component.

Reply:
I'm still unsure how one would intentionally hit this performance drop. I'm running the below test and don't see my memory usage going above 1.3GB 
```cs
        [Test]
        public void TestManyNext()
        {
            AddStep("Add overlay", () =>
            {
                if (!Children.ToList().Contains(nowPlayingOverlay))
                    Add(nowPlayingOverlay);
            });
            AddStep("Remove overlay", () => {
                Remove(nowPlayingOverlay);
            });
            for (int i = 0; i < 1000; i++)
            {
                currentBeatmap = Beatmap.Value;
                AddStep("next track", () => musicController.NextTrack());
                AddUntilStep("track changed", () => Beatmap.Value != currentBeatmap);
            }
        }
```

Reply:
if you can't figure it out please just understand the comments and try and understand the underlying logic. you can google/youtube on reference leaks and how they can occur if you want to learn more.